### DISCUSSION TITLE
Will disabling WebGL enhance privacy?

### DISCUSSION THREAD

**Discussion by ghost** [bromite/bromite#955] [Q&A] [answered]
Disabling WebGL seems to improve fingerprinting mitigations but I am not sure

**Comment by chromer030** [comment]
Isn't it better to hide real WebGL fingerprint by reporting a random fake value instead of fully disabling this feature ?
According to many tech blogs, completely blocking WebGL API is not a good idea, therefore reporting a fake fingerprint could be the best solution to better protect your privacy.

**Reply by ghost** [reply]
I think faking WebGL fingerprinting is a WIP for a while now, there's an issue open on that. In the meantime I want to ask if disabling WebGL is the best thing to do or not

**Reply by chromer030** [reply]
Everyone who is concerned about privacy and highest possible safety should disable it because of issues in fingerprints leaking and some GPU infections and vulnerabilities,There is some rare websites could use webgl for their main functionality though.
We should note that the the risks aren't too great because of active efforts in mitigating bugs and issues by large companies like Google and Mozilla.

**Reply by uazo** [reply]
> completely blocking WebGL API

there is also another way, disable the reading of the canvas from js, through this flag https://source.chromium.org/chromium/chromium/src/+/master:content/public/common/content_switches.cc;l=219;drc=1efce20b143216fbac0a9192b48da7fa834c0ab5;bpv=1;bpt=1, but I warn you, there are many sites that will not work. @csagan5 I think you are well versed on the subject, what do you think?

ah, of course unfortunately in android it is not possible for the user to directly disable...

**Reply by csagan5** [reply]
@uazo you can also have a non-WebGL canvas; yes, disabling that will cause some breakage, that is why the canvas randomisation patch was developed

> unfortunately in android it is not possible for the user to directly disable

We could make it a flag, like for the WebGL one

**Reply by uazo** [reply]
a flag and a content settings is better

**Reply by csagan5** [reply]
Yes, that is the approach used by some other browser too, and asking the user to confirm whether they want to allow the website or not.

Upstream was working on something related, an "entropy budget", I wonder what's the status on that.

**Comment by csagan5** [comment, marked as answer]
> Disabling WebGL seems to improve fingerprinting mitigations but I am not sure

Yes, but also break some websites that either use it to deliver some functionality or use it to do fingerprinting; https://github.com/bromite/bromite/issues/131 would help with the latter.
